File: 2022.01.15/javascript/1065.js
const fs = require('fs');
const n = Number(fs.readFileSync('/dev/stdin'));
let cnt = 0;
let solution = function(n){
    if (n<100){
        return true;
    }
    let prev = n%10;
    n = parseInt(n/10);
    let dif = prev-n%10;
    while (n){
        if ((prev-n%10) !== dif){
            return false;
        }
        prev = n%10;
        n = parseInt(n/10);
    }
    return true;
}
for (let i = 1; i<=n ; i++){
    if(solution(i)){
        cnt++;
    }
}
console.log(cnt);

File: Boj.단계별/6_함수/python/1065.py
n = int(input())
cnt = 0
def hansoo(n):
    if n<100:
        return 1
    prev = n%10
    n //= 10
    dif = n%10 - prev
    while n:
        if dif != n%10 - prev:
            return 0
        prev = n%10
        n //= 10
    return 1
for i in range(1,n+1):
    cnt += hansoo(i)
print(cnt)